tests/integration/awslambda/test_lambda.py

```diff
@@ -798,6 +800,50 @@ def test_large_payloads(self, caplog, lambda_client, create_lambda_function, sna
         result_data = json.loads(to_str(result_data))
         assert payload == result_data
 
+    @pytest.mark.aws_validated
+    @pytest.mark.skip_snapshot_verify  # skipped - the diff is too big
+    @pytest.mark.skipif(
+        os.environ.get("TEST_TARGET") != "AWS_CLOUD",
+        reason="Lambda function state pending for small lambdas not supported on localstack",
+    )
+    def test_function_state(self, lambda_client, lambda_su_role, snapshot):
+        """Tests if a lambda starts in state "Pending" but moves to "Active" at some point"""
+        snapshot.add_transformer(snapshot.transform.lambda_api())
+        # necessary due to changing zip timestamps. We might need a reproducible archive method for this
+        snapshot.add_transformer(snapshot.transform.key_value("CodeSha256"))
+        function_name = f"test-function-{short_uid()}"
+        zip_file = create_lambda_archive(load_file(TEST_LAMBDA_PYTHON_ECHO), get_content=True)
+
+        try:
+
+            def create_function():
+                return lambda_client.create_function(
+                    FunctionName=function_name,
+                    Runtime="python3.9",
+                    Handler="handler.handler",
+                    Role=lambda_su_role,
+                    Code={"ZipFile": zip_file},
+                )
+
+            response = retry(create_function, sleep=3, retries=5)
```

Comment:
Out of curiosity - is this retry required here? Under which circumstances would we expect the `create_function` call to fail initially, but then succeed after some retries?

Reply:
Its a retry our create_lambda_function fixture already has implemented: it can happen due to the propagation times of IAM, so the lambda does not have permissions to assume the role yet. Sadly, there is really no other way then retrying, since we cannot check if the IAM policy which allows lambda to assume the role has propagated yet.

Reply:
Gotcha, thanks for clarifying! 👍